lib/pytap13.py

```diff
@@ -174,13 +169,13 @@ def _parse(self, source):
         if len(self.tests) != self.tests_planned:
             comment = 'DIAG: Expected %s tests, got %s' % \
                 (self.tests_planned, len(self.tests))
-            self.tests.append(Test('not ok', len(self.tests), comment=comment))
+            self.tests.append(Test(b'not ok', len(self.tests), comment=comment))
 
     def parse(self, source):
         if isinstance(source, string_types):
-            self._parse(StringIO(source))
+            self._parse(BytesIO(source))
```

Comment:
(Not a real review comment, just note. I'll give more thoughs a bit later.)

`string_types` is `bytes` and `unicode` on Python 2 and just `str` (unicode string) on Python 3. Let's consider Python 3 case. BytesIO accepts only `bytes`, not `str`:

```python
>>> from io import BytesIO
>>> BytesIO('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: a bytes-like object is required, not 'str'
```

So the if-condition (`isinstance(source, string_types)`) contradicts with the `BytesIO(source)` expression.

Reply:
(In fact, after the patch, we never going into this branch, so it becomes a dead code.)

Reply:
Since we decided to raise a good error instead of support of non UTF-8 data in TAP13 parser, this discussion is not actual anymore.